msal/application.py

```diff
@@ -161,14 +163,18 @@ def __init__(
         self.client_id = client_id
         self.client_credential = client_credential
         self.client_claims = client_claims
-        self.verify = verify
-        self.proxies = proxies
+        if http_client:
+            self.http_client = http_client
+        else:
+            self.http_client = requests.Session()
+            self.http_client.verify = verify
+            self.http_client.proxies = proxies
```

Comment:
We mentioned this in a daily sync-up the other day, that the timeout would also have to be handle differently, i.e. solely via http_client. See also [here](https://github.com/AzureAD/microsoft-authentication-library-for-python/compare/oauth2-session?expand=1#diff-ecd2082a45061b914600fa860e2d38bbR84). (From my side, I can go ahead and further change the "session" keyword in that file, so that you could just copy-and-paste it into this PR, and then focus on modifying other relevant things here).